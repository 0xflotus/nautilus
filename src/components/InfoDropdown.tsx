/**
 * ************************************
 *
 * @module  InfoDropdown.tsx
 * @author
 * @date 3/11/20
 * @description Dropdown display to show categories of service info
 *
 * ************************************
 */
import React from 'react';

type Props = {};

const InfoDropdown: React.FC<Props> = props => {
  return (
    <div className="info-dropdown">
      <div></div>
    </div>
  );
};

export default InfoDropdown;
