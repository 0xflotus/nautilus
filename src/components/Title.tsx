import * as React from 'react';
import Image from 'react-bootstrap/Image';

type Props = {
  projectName: string;
}

const Title: React.FC<Props> = props => (
  <div className="title">

    <Image src="./src/styles/shell.png" fluid />
    <h1>{props.projectName}</h1>
  </div>
)

export default Title;