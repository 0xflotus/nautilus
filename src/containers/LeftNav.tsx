/**
 * ************************************
 *
 * @module  LeftNav.tsx
 * @author 
 * @date 3/11/20
 * @description 
 * 
 * ************************************
 */
import React from 'react';

type Props = {

}

export const LeftNav: React.FC<Props> = props => {

  return (
    <div>
      
    </div>
  );
}
