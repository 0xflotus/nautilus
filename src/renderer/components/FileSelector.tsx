/**
 * ************************************
 *
 * @module  FileSelector.tsx
 * @author Mike D
 * @date 3/11/20
 * @description Button to allow user to upload docker-compose file
 *
 * ************************************
 */
import React from 'react';
import { FaUpload } from 'react-icons/fa';

import { FileUpload } from '../App.d';

type Props = {
  fileUpload: FileUpload;
};

const FileSelector: React.FC<Props> = ({ fileUpload }) => {
  return (
    <div className="file-upload">
      <label htmlFor="files">
        <div className="select-file-button upload-flex">
          <FaUpload className="upload-button" size={24} />
          <h5>Upload</h5>
        </div>
      </label>
      <input
        id="files"
        type="file"
        name="yaml"
        accept=".yml,.yaml"
        style={{ display: 'none' }}
        onChange={(event: React.SyntheticEvent<HTMLInputElement>) => {
          // make sure there was something selected
          if (event.currentTarget) {
            // make sure user uploaded a file
            if (event.currentTarget.files) {
              // fire fileUpload function on first file uploaded
              fileUpload(event.currentTarget.files[0]);
            }
          }
        }}
      />
    </div>
  );
};
export default FileSelector;
