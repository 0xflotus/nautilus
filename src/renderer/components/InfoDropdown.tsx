/**
 * ************************************
 *
 * @module  InfoDropdown.tsx
 * @author Aris Razuri, not danny
 * @date 3/11/20
 * @description Dropdown display to show categories of service info
 *
 * ************************************
 */
import React from 'react';
import { FaAngleDown } from 'react-icons/fa';
import Accordion from 'react-bootstrap/Accordion';
import Card from 'react-bootstrap/Card';

// import { Service } from '../App.d';

type ReactProps = {
  service?: any;
};

type DockerComposeProperties = {
  [prop: string]: string;
};

type ServiceOverview = {
  [prop: string]: any;
};

// type Build = {
//   [prop: string]: string | {};
// };

type context = {
  [prop: string]: string;
}

type EnvironmentVariables = {
  [prop: string]: string;
};

const InfoDropdown: React.FC<ReactProps> = ({ service }) => {
  // Create an object to house text intros for each docker-compose property
  const dockerComposeProperties: DockerComposeProperties = {
    build: 'Build: ',
    context: 'Context: ',
    dockerfile: 'Dockerfile: ',
    args: 'Args: ',
    cache_from: 'Cache_from: ',
    labels: 'Labels: ',
    shm_size: 'Shm_size: ',
    target: 'Target: ',
    image: 'Image: ',
    command: 'Command: ',
    environment: 'Environment: ',
    env_file: 'Env_file: ',
    ports: 'Ports: ',
  };

  // Objects to hold filtered 1D service properties
  const serviceOverview: ServiceOverview = {};

  // Arrays/Objects to hold filtered 2D service properties
  const context: Context = 
  const environmentVariables: EnvironmentVariables = {};

  // if service exists
  if (service) {
    // loop through the properties in each service
    Object.keys(service).forEach((key: string) => {
      // and if the service property can be found in the list of docker-compose properties...
      if (dockerComposeProperties[key]) {
        // ASIDE: if any of the below keys equal the specified strings,
        if (key === 'environment') {
          // ASIDE: set the key at the 2d arrays/objects between lines 55 and x, to be the service's property and set it equal to service's property
          environmentVariables[key] = key;
          console.log(environmentVariables);
        }
        // ...then set the key in the serviceOverview's object to equal the service property and its value to be the service property value
        // ie {Build: ./result}
        serviceOverview[key] = service[key];
        // if the service property is build and it's value is an object,
        if (key === 'build' && typeof service[key] === 'object') {
          // set the key in the serviceOverview object to 'build' and set the value to an empty object
          serviceOverview[key] = '';
        }
      }
    });
  }

  return (
    <div className="info-dropdown">
      <Accordion>
        {/* OVERVIEW */}
        <Card>
          <Card.Header>
            <Accordion.Toggle as={Card.Header} eventKey="0">
              Overview <FaAngleDown />
            </Accordion.Toggle>
          </Card.Header>
          <Accordion.Collapse eventKey="0">
            <Card.Body>
              {Object.keys(serviceOverview).length === 0
                ? 'There are no overview details in your Docker-Compose file'
                : Object.keys(serviceOverview).map(
                    el =>
                      `${dockerComposeProperties[el]}${serviceOverview[el]}`,
                  )}
            </Card.Body>
          </Accordion.Collapse>
        </Card>
      </Accordion>
    </div>
  );
};

export default InfoDropdown;
