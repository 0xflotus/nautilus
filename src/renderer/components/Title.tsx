import * as React from 'react';
import Image from 'react-bootstrap/Image';
import { getStatic } from '../helpers/static';

type Props = {
  // projectName: string;
};

const Title: React.FC<Props> = props => (
  <div className="title">
    <Image src={getStatic('shell.png')} fluid />
    <h1>Nautilus</h1>
  </div>
);

export default Title;
