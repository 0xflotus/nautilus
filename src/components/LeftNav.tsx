/**
 * ************************************
 *
 * @module  LeftNav.tsx
 * @author
 * @date 3/11/20
 * @description
 *
 * ************************************
 */
import React from 'react';

// IMPORT REACT COMPONENTS
import ServiceInfoDisplay from './ServiceInfoDisplay';
import Title from './Title';
import Button from 'react-bootstrap/Button';
import FileSelector from './FileUpload';

type Props = {
  projectName: string;
};

const LeftNav: React.FC<Props> = ({ projectName }) => {
  return (
    <div className="left-nav">

      <Title projectName={projectName} />
      <FileSelector />
      <ServiceInfoDisplay />
    </div>
  );
};

export default LeftNav;
