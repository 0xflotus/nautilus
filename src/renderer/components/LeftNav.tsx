/**
 * ************************************
 *
 * @module  LeftNav.tsx
 * @author
 * @date 3/11/20
 * @description
 *
 * ************************************
 */
import React from 'react';

// IMPORT REACT COMPONENTS
import InfoDropdown from './InfoDropdown';
import FileSelector from './FileUpload';
import Title from './Title';
import { FileUpload, Service } from '../App.d';

type Props = {
  service: Service;
  selectedContainer: string;
  fileUpload: FileUpload;
  fileUploaded: boolean;
};

const LeftNav: React.FC<Props> = ({
  fileUpload,
  fileUploaded,
  selectedContainer,
  service,
}) => {
  return (
    <div className="left-nav">
      <div className="top-half">
        <Title />
        {fileUploaded ? <FileSelector fileUpload={fileUpload} /> : null}
      </div>
      <InfoDropdown selectedContainer={selectedContainer} service={service} />
    </div>
  );
};

export default LeftNav;
