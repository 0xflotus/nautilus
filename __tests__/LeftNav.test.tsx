import React from 'react';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LeftNav from '../src/renderer/components/LeftNav';
import Title from '../src/renderer/components/Title';
import FileSelector from '../src/renderer/components/FileSelector';
import InfoDropdown from '../src/renderer/components/InfoDropdown';

configure({ adapter: new Adapter() });

const props = {
  fileUpload: jest.fn(() => {}),
  fileUploaded: false,
  selectedContainer: '',
  service: {},
};

describe('test the functionality of LeftNav component', () => {
  // Test Outer div
  it('Should render a div with the class of `left-nav`', () => {
    const wrapper = shallow(<LeftNav {...props} />);
    expect(wrapper.find('div.left-nav')).toHaveLength(1);
  });

  // Test top-half div
  it('Should render a div inside of `div.left-nav` with a class of `ltop-half`', () => {
    const wrapper = shallow(<LeftNav {...props} />);
    expect(wrapper.find('div.left-nav').find('div.top-half')).toHaveLength(1);
  });

  // Test Title
  it('Should render a Title component inside of `div.top-half`', () => {
    const wrapper = shallow(<LeftNav {...props} />);
    expect(wrapper.find('div.top-half').find(Title)).toHaveLength(1);
  });

  // Test FileUploaded
  it('`div.top-half` should only have one child if fileUploaded is false', () => {
    const wrapper = shallow(<LeftNav {...props} />);
    expect(wrapper.find('div.top-half').children()).toHaveLength(1);
  });

  it('`div.top-half` should have two children if fileUploaded is true', () => {
    const wrapper = shallow(<LeftNav {...props} fileUploaded={true} />);
    expect(wrapper.find('div.top-half').children()).toHaveLength(2);
  });

  it('Should render a file selector if fileUploaded is true', () => {
    const wrapper = shallow(<LeftNav {...props} fileUploaded={true} />);
    expect(wrapper.find('div.top-half').find(FileSelector)).toHaveLength(1);
  });

  // Test InfoDropdown
  it('Should render an InfoDropdown component inside of `div.left-nav`', () => {
    const wrapper = shallow(<LeftNav {...props} />);
    expect(wrapper.find('div.left-nav').find(InfoDropdown)).toHaveLength(1);
  });
});
