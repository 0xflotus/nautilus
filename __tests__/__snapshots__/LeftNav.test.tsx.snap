// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the functionality of LeftNav component renders correctly 1`] = `
<div
  className="left-nav"
>
  <div
    className="top-half"
  >
    <div
      className="title"
    >
      <img
        src="http://localhost/nautilus_logo.svg"
      />
      <h1>
        Nautilus
      </h1>
    </div>
    <div
      className="file-open"
    >
      <label
        htmlFor="files"
      >
        <div
          className="select-file-button open-flex"
        >
          <svg
            className="open-button"
            fill="currentColor"
            height={24}
            size={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
            />
          </svg>
          <h5>
            Open
          </h5>
        </div>
      </label>
      <input
        accept=".yml,.yaml"
        id="files"
        name="yaml"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
    </div>
  </div>
  <div
    className="info-dropdown"
  >
    <h3>
      Db
    </h3>
    <div
      className="content-wrapper"
    >
      <div
        className="overflow-container"
      >
        <div
          className="overview"
        >
          <div>
            <span
              className="command"
            >
              Image: 
            </span>
            <span
              className="command-values"
            >
              postgres
            </span>
          </div>
          <div>
            <span
              className="command"
            >
              Environment: 
            </span>
            <span
              className="command-values"
            >
              <ul>
                <li
                  className="second-level"
                >
                  <span>
                    POSTGRES_MULTIPLE_DATABASES
                    :
                  </span>
                   
                  dpc_attribution,dpc_queue,dpc_auth,dpc_consent
                </li>
                <li
                  className="second-level"
                >
                  <span>
                    POSTGRES_USER
                    :
                  </span>
                   
                  postgres
                </li>
                <li
                  className="second-level"
                >
                  <span>
                    POSTGRES_PASSWORD
                    :
                  </span>
                   
                  dpc-safe
                </li>
              </ul>
            </span>
          </div>
          <div>
            <span
              className="command"
            >
              Ports: 
            </span>
            <span
              className="command-values"
            >
              <ul>
                <li
                  className="second-level"
                >
                  5432:5432
                </li>
              </ul>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
